File: C0DEF0RCES/2022-12 (december)/1770 c.py
from math import gcd

def solve(a):
  g = a[0]
  for i in range(1, len(a)):
    g = gcd(g, a[i])

  # gcd 1 hole yes
  if g == 1:
    return "YES"

#lcm ber kora
  llcm = a[0]
  for i in range(1, len(a)):
    llcm = (lcm * a[i]) // gcd(lcm, a[i])

  if llcm > 1:
    return "YES"
  else:
    return "NO"

File: C0DEF0RCES/2022-12 (december)/1770 c.cpp
#include <bits/stdc++.h>

using namespace std;

string solve(vector<long long> a) {
  // Find the GCD of all the numbers in the array
  long long g = a[0];
  for (int i = 1; i < a.size(); i++) {
    g = __gcd(g, a[i]);
  }

  // If the GCD is 1, then we can return "YES"
  if (g == 1) {
    return "YES";
  }

  // Find the LCM of all the numbers in the array
  long long lcm = a[0];
  for (int i = 1; i < a.size(); i++) {
    lcm = (lcm * a[i]) / __gcd(lcm, a[i]);
  }

  // If the LCM is greater than 1, then the GCD of all the numbers
  // in the array plus the LCM will be 1
  if (lcm > 1) {
    return "YES";
  } else {
    return "NO";
  }
}

int main() {
  int t;
  cin >> t;
  while (t--) {
    int n;
    cin >> n;
    vector<long long> a(n);
    for (int i = 0; i < n; i++) {
      cin >> a[i];
    }
    cout << solve(a) << endl;
  }
  return 0;
}
